lib/puppet_x/relay/agent/backend/orchestrator.rb

```diff
@@ -45,7 +45,14 @@ def exec(run, _state_dir, schedule)
               message: e.message,
               body: e.response.body,
             })
-            raise
+            run_results = {
+              error: 'Failed to send request to orchestrator API',
+              message: e.message,
+              response: JSON.parse(e.response.body),
```

Comment:
You need to probably catch `JSON::ParserError` around this in case something Really Bad(tm) happens.

Reply:
I've added a `rescue` on the line, assigning the value without parsing, hope that's enough?